From 22cb334f0eeb4084d28dc73839bd138ce89f0912 Mon Sep 17 00:00:00 2001
From: Eddie Huang <eddie.huang@mediatek.com>
Date: Fri, 30 Jan 2015 16:00:17 +0800
Subject: [PATCH 8/8] Fix: clk: Fix CCF cause kernel run to shell fail issue

Still checking which clock be disable accidentially.
This is a workaround solution.

Change-Id: I260b27811fd378a2f53287a4b563c8d9e9c4e62e
Signed-off-by: Eddie Huang <eddie.huang@mediatek.com>
---
 drivers/clk/mediatek/clk-gate.c | 2 +-
 drivers/clk/mediatek/clk-mtk.c  | 4 ++--
 drivers/clk/mediatek/clk-pll.c  | 1 +
 3 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/clk/mediatek/clk-gate.c b/drivers/clk/mediatek/clk-gate.c
index f05507a..c9bc115 100644
--- a/drivers/clk/mediatek/clk-gate.c
+++ b/drivers/clk/mediatek/clk-gate.c
@@ -119,7 +119,7 @@ struct clk *mtk_clk_register_gate(
 		return ERR_PTR(-ENOMEM);
 
 	init.name = name;
-	init.flags = CLK_SET_RATE_PARENT;
+	init.flags = CLK_IGNORE_UNUSED;
 	init.parent_names = parent_name ? &parent_name : NULL;
 	init.num_parents = parent_name ? 1 : 0;
 	init.ops = ops;
diff --git a/drivers/clk/mediatek/clk-mtk.c b/drivers/clk/mediatek/clk-mtk.c
index ce4f2ac..288cca4 100644
--- a/drivers/clk/mediatek/clk-mtk.c
+++ b/drivers/clk/mediatek/clk-mtk.c
@@ -33,7 +33,7 @@ void mtk_init_factors(struct mtk_fixed_factor *clks, int num,
 		struct mtk_fixed_factor *ff = &clks[i];
 
 		clk = clk_register_fixed_factor(NULL, ff->name, ff->parent_name,
-				CLK_SET_RATE_PARENT, ff->mult, ff->div);
+				0, ff->mult, ff->div);
 
 		if (IS_ERR(clk)) {
 			pr_err("Failed to register clk %s: %ld\n",
@@ -143,7 +143,7 @@ struct clk *mtk_clk_register_mux(
 		&mux->hw, &clk_mux_ops,
 		NULL, NULL,
 		gate_hw, gate_ops,
-		CLK_SET_RATE_PARENT);
+		CLK_IGNORE_UNUSED);
 
 	if (IS_ERR(clk)) {
 		kfree(gate);
diff --git a/drivers/clk/mediatek/clk-pll.c b/drivers/clk/mediatek/clk-pll.c
index 59dee83..1d93897 100644
--- a/drivers/clk/mediatek/clk-pll.c
+++ b/drivers/clk/mediatek/clk-pll.c
@@ -50,6 +50,7 @@ struct clk *mtk_clk_register_pll(
 	pll->hw.init = &init;
 
 	init.name = name;
+	init.flags = CLK_IGNORE_UNUSED;
 	init.ops = ops;
 	init.parent_names = &parent_name;
 	init.num_parents = 1;
-- 
1.9.1

